pandaharvester/harvesterbody/submitter.py

```diff
@@ -318,7 +318,7 @@ def run(self):
                                             and queueConfig.mapType != WorkSpec.MT_MultiWorkers:
                                         workSpecsToEnqueue = \
                                             [[w] for w in workSpecList if w.status
-                                             in (WorkSpec.ST_submitted, WorkSpec.ST_running)]
+                                             in (WorkSpec.ST_submitted, WorkSpec.ST_running, WorkSpec.ST_idle)]
```

Comment:
Is WorkSpec.ST_idle required here and L348?

Reply:
But worker status never change to idle in submitter. E.g. it is set to running, missed, or submitted.

https://github.com/PanDAWMS/panda-harvester/blob/335d1bcc878a8961d3332487c95a82af7d5c4f68/pandaharvester/harvesterbody/submitter.py#L250
https://github.com/PanDAWMS/panda-harvester/blob/335d1bcc878a8961d3332487c95a82af7d5c4f68/pandaharvester/harvesterbody/submitter.py#L272
https://github.com/PanDAWMS/panda-harvester/blob/335d1bcc878a8961d3332487c95a82af7d5c4f68/pandaharvester/harvesterbody/submitter.py#L275

Also it can be pending for new workers, ready/running for existing workers with job late-binding, or one of final state. idle is an active state after submitted, which is set by monitor instead of submitter. For example ST_idle in L321 doesn't do anything. It is harmless to add ST_idle to those lists but it is logically useless. I suggest that you remove it from the lists.